File: ming/0913/Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i >= 0; i--){
            if (digits[i] == 9){
                digits[i] = 0;
            }else{
                digits[i]++;
                return digits;
            }
        }
        
         int[] newDigi = new int[digits.length + 1];
         newDigi[0] = 1;
         for (int i = 1; i < newDigi.length; i++){
            newDigi[i] = 0;
         }
         return newDigi;
    }
}

File: mu_wang/9_13/Plus_One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        result = list()
        n = len(digits)
        num = 0
        for a in range(n):
             num+=digits[a]*(10**(n-1))
             n-=1
        num =num+1
        result = result + [ int(x) for x in str(num)]
        return result




    def plusOne_2(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        for i in range(len(digits)-1,-1,-1):
            if digits[i]!=9:
                digits[i]+=1
                return digits
            else:
                digits[i]=0
                
        digits[0]=1
        digits+=[0]
        return digits